extensions/liquibase/deployment/src/main/java/io/quarkus/liquibase/LiquibaseProcessor.java

```diff
@@ -0,0 +1,319 @@
+            BuildProducer<BeanContainerListenerBuildItem> containerListenerProducer,
+            LiquibaseRecorder recorder,
+            DataSourceInitializedBuildItem dataSourceInitializedBuildItem,
+            BuildProducer<GeneratedBeanBuildItem> generatedBeanBuildItem,
+            BuildProducer<NativeImageResourceBundleBuildItem> resourceBundle) {
+
+        featureProducer.produce(new FeatureBuildItem(FeatureBuildItem.LIQUIBASE));
+
+        AdditionalBeanBuildItem unremovableProducer = AdditionalBeanBuildItem.unremovableOf(LiquibaseProducer.class);
+        additionalBeanProducer.produce(unremovableProducer);
+
+        Collection<String> dataSourceNames = DataSourceInitializedBuildItem
+                .dataSourceNamesOf(dataSourceInitializedBuildItem);
+        new LiquibaseDatasourceBeanGenerator(dataSourceNames, generatedBeanBuildItem).createLiquibaseProducerBean();
+
+        containerListenerProducer.produce(
+                new BeanContainerListenerBuildItem(recorder.setLiquibaseBuildConfig(liquibaseBuildConfig)));
+
+        resourceProducer.produce(
+                new NativeImageResourceBuildItem(getChangeLogs(dataSourceInitializedBuildItem).toArray(new String[0])));
+
+        // liquibase XSD
+        resourceProducer.produce(new NativeImageResourceBuildItem(
+                "www.liquibase.org/xml/ns/dbchangelog/dbchangelog-3.5.xsd",
+                "www.liquibase.org/xml/ns/dbchangelog/dbchangelog-3.6.xsd",
+                "www.liquibase.org/xml/ns/dbchangelog/dbchangelog-3.7.xsd",
+                "www.liquibase.org/xml/ns/dbchangelog/dbchangelog-3.8.xsd",
+                "www.liquibase.org/xml/ns/dbchangelog/dbchangelog-ext.xsd",
+                "www.liquibase.org/xml/ns/pro/liquibase-pro-3.8.xsd",
+                "liquibase.build.properties"));
+
+        // liquibase resource bundles
+        Stream.of("liquibase/i18n/liquibase-core",
+                "com.sun.org.apache.xerces.internal.impl.msg.XMLSchemaMessages",
+                "com.sun.org.apache.xerces.internal.impl.xpath.regex.message",
+                "com.sun.org.apache.xerces.internal.impl.msg.XMLMessages")
+                .map(NativeImageResourceBundleBuildItem::new)
+                .forEach(resourceBundle::produce);
+    }
+
+    /**
+     * Handles all the operations that can be recorded in the RUNTIME_INIT execution time phase
+     *
+     * @param recorder Used to set the runtime config
+     * @param liquibaseRuntimeConfig The Liquibase configuration
+     */
+    @Record(ExecutionTime.RUNTIME_INIT)
+    @BuildStep
+    ServiceStartBuildItem configureRuntimeProperties(LiquibaseRecorder recorder,
+            LiquibaseRuntimeConfig liquibaseRuntimeConfig,
+            BeanContainerBuildItem beanContainer) {
+        try {
+            recorder.configureLiquibaseProperties(liquibaseRuntimeConfig, beanContainer.getValue());
+            recorder.doStartActions(liquibaseRuntimeConfig, beanContainer.getValue());
```

Comment:
You will need something similar to https://github.com/quarkusio/quarkus/pull/6301/files#diff-1fdde7b94eca712fd8aa08fe51e38c39R111 . It was just merged into the Flyway extension, that's normal you missed it.

Reply:
Done